core/src/main/scala/org/apache/gearpump/cluster/client/ClientContext.scala

```diff
@@ -60,11 +60,13 @@ class ClientContext(masters: Iterable[HostPort]) {
 
   def submit(app : Application, jarPath: String) : Int = {
     val client = new MasterClient(master)
+    val appName = checkAndAddNamePrefix(app, System.getProperty(GEARPUMP_APP_NAME_PREFIX))
```

Comment:
checkAndAddNamePrefix actually only requires app.name, but we are passing the whole application.

maybe someting like this is better:
```
checkname(app.name)
val newName = s"$prefix_${app.name}"